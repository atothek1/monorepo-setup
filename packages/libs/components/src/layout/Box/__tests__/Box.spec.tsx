import React from "react";
import { render } from "@mono/testing";
import { Box } from "../index";

describe( "Box snapshots", () => {
  const testId = "testid.box";

  it( "should render default", () => {
    const { getByTestId } = render( <Box testId={ testId }>test</Box> );
    expect( getByTestId( testId ) ).toMatchSnapshot();
  } );

  it( "should render as column", () => {
    const { getByTestId } = render( <Box testId={ testId } column>test</Box> );
    expect( getByTestId( testId ) ).toMatchSnapshot();
  } );

  it( "should render as <main> tag", () => {
    const { getByTestId } = render( <Box testId={ testId } as="main">test</Box> );
    expect( getByTestId( testId ) ).toMatchSnapshot();
  } );
} );
