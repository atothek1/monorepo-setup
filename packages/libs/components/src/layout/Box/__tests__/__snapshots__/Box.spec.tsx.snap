// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box snapshots should render as <main> tag 1`] = `
<main
  class="root"
  data-testid="testid.box"
>
  test
</main>
`;

exports[`Box snapshots should render default 1`] = `
<div
  class="root"
  data-testid="testid.box"
>
  test
</div>
`;

exports[`Box snapshots should render with column classname 1`] = `
<div
  class="root column"
  data-testid="testid.box"
>
  test
</div>
`;

exports[`Box snapshots should render with custom classname 'test' 1`] = `
<div
  class="root test"
  data-testid="testid.box"
>
  test
</div>
`;
